### DISCUSSION TITLE
offline usage

### DISCUSSION THREAD

**Discussion by tobywu6652** [open-webui/open-webui#5759] [General]
# Bug Report

## Environment

- Open WebUI Version: v0.3.30

## Expected Behavior:

Open WebUI can run in an offline (unable to access the Internet) environment.

## Actual Behavior:

Open WebUI cannot run in an offline environment fluently.

## Description

When visiting the site, the API ( /api/version/updates ) will check the version (current & latest) information through the Internet. And before a long timeout, the site will not be displayed.

**Comment by Lodisaq** [comment]
yes，I was deeply puzzled by the bug.  the open webui keeps accessing '/api/version/updates' and doesn't load the page until it times out.

**Comment by gaby** [comment]
There's a lot of things trying to hit the internet when using `Open-WebUI`.
- Version update check, even though its running in Docker.
- When uploading a file in a chat, it tries to download a model from HF.
- The Swagger docs do not work without internet, they dont even render #5714 
- The default embedding if not changed to `ollama` it tries to go to the internet and the UI won't start at all.

Kind of ironic given the description of the project says:

> Open WebUI is an extensible, feature-rich, and user-friendly self-hosted WebUI designed to operate entirely *offline*.

I think adding a ENV variable for "Offline Mode" could help the backend make a decision on wether to reach the internet or not.

**Comment by tjbck** [comment]
By all means, feel free to make a PR! 🙌

**Comment by tobywu6652** [comment]
Is version 0.3.19 the latest version without a version detection API?

**Comment by joosmatt** [comment]
Hi, I was able to prevent the API from checking for updates and causing a timeout by modifying the open-webui/backend/open_webui/main.py file (approx. line 2153)

```
@app.get("/api/version/updates")
async def get_app_latest_release_version():
#    try:
#        async with aiohttp.ClientSession(trust_env=True) as session:
#            async with session.get(
#                "https://api.github.com/repos/open-webui/open-webui/releases/latest"
#            ) as response:
#                response.raise_for_status()
#                data = await response.json()
#                latest_version = data["tag_name"]

#                return {"current": VERSION, "latest": latest_version[1:]}
#    except aiohttp.ClientError:
#        raise HTTPException(
#            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
#            detail=ERROR_MESSAGES.RATE_LIMIT_EXCEEDED,
#        )
    return {"current": VERSION, "latest": VERSION}
```
This way, it returns always the current version as the latest version. 

You can modify the main.py file in a running container, e.g. ./var/lib/docker/overlay2/9bde71c5174c791e76d1e4e19b79dac415938b9412a4e9077bec930c3a4260f6/merged/app/backend/open_webui/main.py 
and restart the container ID after the modification

hope it helps!

**Comment by saravanan30erd** [comment]
@tjbck 

>> By all means, feel free to make a PR!

Thanks, I have created PR https://github.com/open-webui/open-webui/pull/5977 for this original issue. Please check it.

**Reply by gaby** [reply]
Awesome 💪
